src/movegen.cpp

```diff
@@ -223,7 +223,7 @@ namespace {
             target = ~pos.pieces();
             break;
         case EVASIONS:
-            target = between_bb(pos.square<KING>(Us), lsb(pos.checkers())) | pos.checkers();
+            target = ray_bb(pos.square<KING>(Us), lsb(pos.checkers()));
```

Comment:
@snicolet here, if the checker is a knight, we have to mark it, that's the foundation on adding s2 on `RayBB`. it's not a line anymore, it has also trigger when the knight is attacking.

Reply:
it should also be a "ray" for a knight. That's the only place where it needs to include `s2` "no matter what", all other use is indifferent on s2. except on `has_game_cycle`, so there I did a workaround.